qemu_build

```diff
@@ -0,0 +1,165 @@
+#!/bin/bash
+
+set -e
+set -u
+set -o pipefail
+
+BASENAME=bitnami/minideb
+pub_key_dir="$(mktemp -d)"
+
+do_sudo() {
+  if [[ "0" == "$(id --user)" ]]; then
+    "$@"
+  else
+    sudo "$@"
+  fi
+}
+
+do_ssh() {
+  ssh -o "StrictHostKeyChecking=no" -o "UserKnownHostsFile=/dev/null" root@localhost -t -p 5555 -i "$pub_key_dir/id_rsa" "$@"
+}
+
+finish() {
+  echo "Shutting down QEMU..."
+
+  n=0
+  until [ "$n" -ge 15 ]
+  do
+     do_ssh "true" && break
+     n=$((n+1))
+     sleep 30
+  done
+
+  do_ssh "poweroff" || true
+
+  sleep 5
+  n=0
+  until [ "$n" -ge 5 ]
+  do
+     kill -0 "$PID" && break
+     n=$((n+1))
+     sleep 5
+  done
+
+  kill -9 "$PID" || true
+  rm -f "$IMAGE_FILE" "$PIDFILE"
+}
+
+if [[ ! -f /etc/debian_version ]]; then
+  echo "minideb can currently only be built on debian based distros, aborting..."
+  exit 1
+fi
+
+if [ -z "$1" ]; then
+    echo "You must specify the dist to build"
+    exit 1
+fi
+
+DIST=$1
+PLATFORM=${2:-amd64}
+
+while do_sudo fuser /var/{lib/{dpkg,apt/lists},cache/apt/archives}/lock >/dev/null 2>&1; do
+    sleep 1
+done
+
+do_sudo apt-get update
+do_sudo apt-get install -y qemu-kvm libvirt-bin qemu-utils genisoimage virtinst curl rsync qemu-system-x86 qemu-system-arm cloud-image-utils
+
+mkdir -p .kvm-images/{amd64,arm64}
+
+if [[ ! -f .kvm-images/amd64/bionic-server-cloudimg-amd64.img && "$PLATFORM" == "amd64" ]]; then
+  curl -SL https://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-amd64.img > .kvm-images/amd64/bionic-server-cloudimg-amd64.img
```

Comment:
question: do Debian offer equivalent images?

Reply:
Maybe debian openstack images could be used (https://cdimage.debian.org/cdimage/openstack/current/) but I'm not sure about it.

Reply:
I think this is fine given that this is mainly for local testing. It just struck me a bit odd to be using Ubuntu to build Debian.

Reply:
I just replaced the images with Debian ones.